### PR TITLE
[opencascade] Cleanup and fixes

### PR DESCRIPTION
**Pull request by dg0yt**
Cleanup.
Fix mingw.
Enable arm, osx, uwp.
Fix license/copyright.
Disable uncontrolled system Tk dependency.
Revise dependency patches.

### CODE REVIEW

**Comment by dg0yt** [comment]
TIL CMake tracing breaks at 1 GB... but I got enough data to fix the build

File: ports/opencascade/dependencies.patch [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+diff --git a/adm/cmake/occt_toolkit.cmake b/adm/cmake/occt_toolkit.cmake
+index c9399159..f707e00f 100644
+--- a/adm/cmake/occt_toolkit.cmake
++++ b/adm/cmake/occt_toolkit.cmake
+@@ -475,11 +475,11 @@ else()
+   endif()
+ endif()
+ 
+-if (BUILD_SHARED_LIBS OR EXECUTABLE_PROJECT)
++if (1)
```
Comment (dg0yt) [inline ports/opencascade/dependencies.patch:65]: FTR we do want the link libraries also for static linkage:
- for include directories during the build when needed
- for cmake config export.

**Comment by FrankXie05** [comment]
```
./vcpkg install opencascade[core,freetype,tbb]:x64-windows
```
Error:
```
[174/5599] cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_dll --intdir=src\TKernel\CMakeFiles\TKernel.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1437~1.328\bin\Hostx64\x64\link.exe  @CMakeFiles\TKernel.rsp  /out:win64\vc14\bind\TKernel.dll /implib:win64\vc14\libd\TKernel.lib /pdb:win64\vc14\bin\TKernel.pdb /dll /version:7.7 /machine:x64 /nologo    /debug /INCREMENTAL  && cd ."
FAILED: win64/vc14/bind/TKernel.dll win64/vc14/libd/TKernel.lib 
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_dll --intdir=src\TKernel\CMakeFiles\TKernel.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1437~1.328\bin\Hostx64\x64\link.exe  @CMakeFiles\TKernel.rsp  /out:win64\vc14\bind\TKernel.dll /implib:win64\vc14\libd\TKernel.lib /pdb:win64\vc14\bin\TKernel.pdb /dll /version:7.7 /machine:x64 /nologo    /debug /INCREMENTAL  && cd ."
LINK Pass 1: command "C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1437~1.328\bin\Hostx64\x64\link.exe @CMakeFiles\TKernel.rsp /out:win64\vc14\bind\TKernel.dll /implib:win64\vc14\libd\TKernel.lib /pdb:win64\vc14\bin\TKernel.pdb /dll /version:7.7 /machine:x64 /nologo /debug /INCREMENTAL /MANIFEST /MANIFESTFILE:src\TKernel\CMakeFiles\TKernel.dir/intermediate.manifest src\TKernel\CMakeFiles\TKernel.dir/manifest.res" failed (exit code 1104) with the following output:
LINK : fatal error LNK1104: cannot open file 'tbbmalloc.lib'
```
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/13406981/install-x64-windows-dbg-out.log)

**Comment by dg0yt** [comment]
Please retry with new commit. If it fails again, reset the PR to draft.

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**